### PR TITLE
Handle error conditions when simulating ingest pipelines with verbosity enabled

### LINKED ISSUE #63199: [Ingest node pipelines] 502 Bad Gateway error when simulating a pipeline
**Elasticsearch version**: `master`

**Description of the problem including expected versus actual behavior**:
Running the simulate pipeline API with verbose enabled against a pipeline that has a `pipeline` processor, and the pipeline defined is expected to fail with the sample documents provided, results in: `{"statusCode":502,"error":"Bad Gateway","message":"Client request timeout"}`. I would expect the request to be successful and return the error. This works as expected **without** `verbose` enabled.

**Steps to reproduce**:
 1. Create a pipeline:
```
PUT _ingest/pipeline/pipeline_expected_to_fail
{
   "description": "_description",
    "processors": [
      {
        "rename": {
          "field": "dont_exist",
          "target_field": "field1"
        }
      }
    ]
}
```
 2. Create a second pipeline that defines a `pipeline` processor with the pipeline created in step 1. For example:
```
PUT _ingest/pipeline/my_pipeline
{
   "description": "_description",
    "processors": [
      {
        "set" : {
          "field" : "field2",
          "value" : "_value"
        }
      },
      {
        "pipeline" : {
          "name": "pipeline_expected_to_fail"
        }
      }
    ]
}
```
 3. Run the simulate API with verbose enabled against the pipeline created in step 2.
```
POST /_ingest/pipeline/my_pipeline/_simulate?verbose
{
  "docs": [
    {
      "_index": "index",
      "_id": "id",
      "_source": {
        "foo": "bar"
      }
    }
  ]
}
```
4. Note the request results in: `{"statusCode":502,"error":"Bad Gateway","message":"Client request timeout"}`

### PR DESCRIPTION
**Pull request by danhermann**
Fixes #63199.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/ingest/TrackingResultProcessor.java [hunk 1]
```diff
@@ -74,19 +74,15 @@ public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Ex
             }
             ingestDocumentCopy.executePipeline(pipelineToCall, (result, e) -> {
                 // do nothing, let the tracking processors throw the exception while recording the path up to the failure
-                if (e instanceof ElasticsearchException) {
-                    ElasticsearchException elasticsearchException = (ElasticsearchException) e;
-                    //else do nothing, let the tracking processors throw the exception while recording the path up to the failure
-                    if (elasticsearchException.getCause() instanceof IllegalStateException) {
```
[the remark this answers is not in the document]
Comment (danhermann) [reply server/src/main/java/org/elasticsearch/ingest/TrackingResultProcessor.java:80]: Additionally, it seems brittle to check for pipeline cycles based only on the class of the inner exception when there's nothing that would prevent other processors from throwing exceptions of type `IllegalStateException` that have nothing to do with pipeline cycles.

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/1

File: server/src/main/java/org/elasticsearch/ingest/TrackingResultProcessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakelandis) [reply server/src/main/java/org/elasticsearch/ingest/TrackingResultProcessor.java:80]: This was briefly discussed a while back https://github.com/elastic/elasticsearch/pull/34155#discussion_r222502476 ..IIRC it had to do with amount of overhead to introduce a custom exception for an edge case.

**Review by jakelandis — APPROVED** [review]
LGTM - thanks for fixing this. However a [REST test](https://github.com/elastic/elasticsearch/blob/master/modules/ingest-common/src/yamlRestTest/resources/rest-api-spec/test/ingest/90_simulate.yml) would be a welcome addition.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
I added a REST test and narrowed the conditions used to identify the pipeline cycle error.

**Pull request merged**